source/libraries/ContractDeployer.ts

```diff
@@ -2,48 +2,117 @@
+    private contracts;
+    private signatures;
+    private bytecodes;
+    private utils;
+    private constants;
     private gasAmount;
+    private testAccountSecretKeys;
+    private testAccounts;
     private controller;
+    private universe;
+    private cash;
+    private binaryMarket;
+    private categoricalMarket;
+    private scalarMarket;
 
-    public getUploadedContracts() {
-        return this.uploadedContracts;
-    }
-
-    public async initialize(eth: Eth, contractJson: string, fromAddress: string, gasAmount: number): Promise<boolean> {
-        this.eth = eth;
+    public constructor(ethjsQuery: EthjsQuery, contractJson: string, gasAmount: number, secretKeys: string[]) {
+        this.ethjsQuery = ethjsQuery;
+        this.ethjsContract = new EthjsContract(ethjsQuery);
         this.compiledContracts = JSON.parse(contractJson);
-        this.uploadedContracts = {};
-        this.fromAddress = fromAddress;
+        this.signatures = [];
+        this.bytecodes = [];
+        this.contracts = [];
         this.gasAmount = gasAmount;
+        this.testAccountSecretKeys = secretKeys;
+    }
+
+    public async deploy(): Promise<boolean> {
+        this.testAccounts = await generateTestAccounts(this.testAccountSecretKeys);
 
         this.controller = await this.upload("../source/contracts/Controller.sol");
         const ownerAddress = (await this.controller.owner())[0];
-        if (ownerAddress !== fromAddress) {
+        if (ownerAddress.toLowerCase() !== this.testAccounts[0].address) {
             throw new Error("Controller owner does not equal from address");
         }
-
         await this.uploadAllContracts();
         await this.whitelistTradingContracts();
         await this.initializeAllContracts();
         await this.approveCentralAuthority();
+        this.universe = await this.createGenesisUniverse();
+        this.cash = await this.getSeededCash();
+        // TODO: Make sure utils is getting uploaded correctly
+        // this.utils = await this.upload("../tests/solidity_test_helpers/Utils.sol");
+        this.binaryMarket = await this.createReasonableBinaryMarket(this.universe, this.cash);
+        // TODO: Make sure catigorical & scalar markets are getting created correctly
+        // this.categoricalMarket = this.createReasonableCategoricalMarket(this.universe, 3, this.cash);
+        // this.scalarMarket = this.createReasonableScalarMarket(this.universe, 40, this.cash);
+        // TODO: Make sure constants is getting uploaded correctly
+        // this.constants = await this.upload("../tests/solidity_test_helpers/Constants.sol");
 
         return true;
     }
 
-    public async uploadAndAddDelegatedToController(contractFileName: string, contractName: string): Promise<ContractBlockchainData|undefined> {
+    // Helper functions
+
+    public async parseBlockTimestamp(blockTimestamp): Promise<Date> {
+        const timestampHex = "0x" + JSON.stringify(blockTimestamp).replace(/\"/g, "");
```

Comment:
Recommend using template strings rather than concatenation:
```
`0x${JSON.stringify(blockTimestamp).replace(/\"/g, "")}`;
```

Not a big deal in this case, but I find it a generally good pattern to follow.

Reply:
Ah, good idea.  Will change that as well.